### PR TITLE
ad_spdif: handle const buf pointee in avio_alloc_context

### PR DESCRIPTION
**Pull request by Dudemanguy**
ffmpeg recently changed this field to be const which causes our CI to fail on newer versions.

See: https://github.com/FFmpeg/FFmpeg/commit/2a68d945cd74265bb71c3d38b7a2e7f7d7e87be5

### CODE REVIEW

**Review by kasper93 — APPROVED** [review]

**Comment by RoverAz** [comment]
Getting a runtime error after this update

[ffmpeg] SWR: Input channel layout "" is invalid or unsupported.
[swresample] Cannot open Libavresample context.
[swresample] libswresample failed to initialize.
Cannot convert decoder/filter output to any format supported by the output.

**Comment by Dudemanguy** [comment]
There's further changes required apparently. This only fixed building with ffmpeg master.

File: audio/decode/ad_spdif.c [hunk 1]
```diff
@@ -59,7 +59,11 @@ struct spdifContext {
     struct mp_decoder public;
 };
 
+#if LIBAVCODEC_VERSION_MAJOR < 61
 static int write_packet(void *p, uint8_t *buf, int buf_size)
+#else
+static int write_packet(void *p, const uint8_t *buf, int buf_size)
```
[the remark this answers is not in the document]
Comment (kasper93) [reply audio/decode/ad_spdif.c:65]: function pointers need to have exact type

[the remark this answers is not in the document]
Comment (Dudemanguy) [reply audio/decode/ad_spdif.c:65]: It was an error either way if that's what you're asking. const to the non-const field (old ffmpeg) and non-const to the new const field (new ffmpeg).

**Pull request merged**